reference-implementation/__tests__/composition.js

```diff
@@ -0,0 +1,367 @@
+'use strict';
+const { URL } = require('url');
+const { parseFromString } = require('../lib/parser.js');
+const { appendMap } = require('../lib/composer.js');
+
+const mapBaseURL = new URL('https://example.com/app/index.html');
+
+function composeMaps(mapLikes) {
+  if (!Array.isArray(mapLikes) || mapLikes.length < 1) {
+    throw new Error('composeMaps must be given a non-empty array of mapLikes');
+  }
+  let map = parseFromString(JSON.stringify(mapLikes.shift()), mapBaseURL);
+  for (const mapLike of mapLikes) {
+    const newMap = parseFromString(JSON.stringify(mapLike), mapBaseURL);
+    map = appendMap(map, newMap);
+  }
+  return JSON.parse(JSON.stringify(map));
```

Comment:
What is the purpose of this? It seems like it's going to hide something from us...

Reply:
It was left over from when we had URL objects in the result, as a way to make it easier to write tests. Removed now.